### PR TITLE
Add docs directory

### PR DESCRIPTION
**Pull request by alessio**

### CODE REVIEW

File: docs/cosmos-hub/deploy-testnet.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+- [Install docker](https://docs.docker.com/engine/installation/)
+- [Install docker-compose](https://docs.docker.com/compose/install/)
+
+### Build
+
+Build the `gaiad` binary (linux) and the `tendermint/gaiadnode` docker image required for running the `localnet` commands. This binary will be mounted into the container and can be updated rebuilding the image, so you only need to build the image once.
+
+```bash
+# Work from the SDK repo
+cd $GOPATH/src/github.com/cosmos/cosmos-sdk
```
Comment (fedekunze) [inline docs/cosmos-hub/deploy-testnet.md]: this should be `gaia`, right?

File: docs/cosmos-hub/gaiacli.md [hunk 2]
```diff
@@ -0,0 +1,910 @@
...
+
+::: tip Note
+
+The action tag always equals the message type returned by the `Type()` function of the relevant message.
+
+You can find a list of available `tags` on each of the SDK modules:
+
+- [Common tags](https://github.com/cosmos/cosmos-sdk/blob/d1e76221d8e28824bb4791cb4ad8662d2ae9051e/types/tags.go#L57-L63)
+- [Staking tags](https://github.com/cosmos/cosmos-sdk/blob/d1e76221d8e28824bb4791cb4ad8662d2ae9051e/x/staking/tags/tags.go#L8-L24)
+- [Governance tags](https://github.com/cosmos/cosmos-sdk/blob/d1e76221d8e28824bb4791cb4ad8662d2ae9051e/x/gov/tags/tags.go#L8-L22)
```
Comment (fedekunze) [inline docs/cosmos-hub/gaiacli.md]:
```suggestion
- [Governance tags](https://cosmos.network/docs/spec/governance/04_tags.html#tags)
```

File: docs/cosmos-hub/gaiacli.md [hunk 3]
```diff
@@ -0,0 +1,910 @@
...
+- [Slashing tags](https://github.com/cosmos/cosmos-sdk/blob/d1e76221d8e28824bb4791cb4ad8662d2ae9051e/x/slashing/handler.go#L52)
+- [Distribution tags](https://github.com/cosmos/cosmos-sdk/blob/develop/x/distribution/tags/tags.go#L8-L17)
+- [Bank tags](https://github.com/cosmos/cosmos-sdk/blob/d1e76221d8e28824bb4791cb4ad8662d2ae9051e/x/bank/keeper.go#L193-L206)
```
Comment (fedekunze) [inline docs/cosmos-hub/gaiacli.md]:
```suggestion
- [Bank tags](https://cosmos.network/docs/spec/bank/04_tags.html#tags)
```

File: docs/cosmos-hub/gaiacli.md [hunk 4]
```diff
@@ -0,0 +1,910 @@
...
+```
+
+::: tip Note
+
+The action tag always equals the message type returned by the `Type()` function of the relevant message.
+
+You can find a list of available `tags` on each of the SDK modules:
+
+- [Common tags](https://github.com/cosmos/cosmos-sdk/blob/d1e76221d8e28824bb4791cb4ad8662d2ae9051e/types/tags.go#L57-L63)
+- [Staking tags](https://github.com/cosmos/cosmos-sdk/blob/d1e76221d8e28824bb4791cb4ad8662d2ae9051e/x/staking/tags/tags.go#L8-L24)
```
Comment (fedekunze) [inline docs/cosmos-hub/gaiacli.md]:
```suggestion
- [Staking tags](https://cosmos.network/docs/spec/staking/06_tags.html#tags)
```

File: docs/cosmos-hub/gaiacli.md [hunk 5]
```diff
@@ -0,0 +1,910 @@
...
+- [Governance tags](https://github.com/cosmos/cosmos-sdk/blob/d1e76221d8e28824bb4791cb4ad8662d2ae9051e/x/gov/tags/tags.go#L8-L22)
+- [Slashing tags](https://github.com/cosmos/cosmos-sdk/blob/d1e76221d8e28824bb4791cb4ad8662d2ae9051e/x/slashing/handler.go#L52)
```
Comment (fedekunze) [inline docs/cosmos-hub/gaiacli.md]:
```suggestion
- [Slashing tags](https://cosmos.network/docs/spec/slashing/06_tags.html#tags)
```

File: docs/cosmos-hub/gaiacli.md [hunk 6]
```diff
@@ -0,0 +1,910 @@
...
+- [Distribution tags](https://github.com/cosmos/cosmos-sdk/blob/develop/x/distribution/tags/tags.go#L8-L17)
```
Comment (fedekunze) [inline docs/cosmos-hub/gaiacli.md]:
```suggestion
- [Distribution tags](https://cosmos.network/docs/spec/distribution/06_tags.html)
```

File: docs/cosmos-hub/gaiacli.md [hunk 7]
```diff
@@ -0,0 +1,910 @@
...
+
+##### Query Delegations To Validator
+
+You can also query all of the delegations to a particular validator:
+
+```bash
+  gaiacli query delegations-to <account_cosmosval>
+```
+
+### Governance
```
Comment (fedekunze) [inline docs/cosmos-hub/gaiacli.md]: isn't this intro SDK specific ?

File: docs/cosmos-hub/genesis.md [hunk 8]
```diff
@@ -0,0 +1,332 @@
+# Genesis File
```
Comment (fedekunze) [inline docs/cosmos-hub/genesis.md]: imo the SDK docs should keep everything on this file but the module specific genesis states

File: docs/cosmos-hub/hd-wallets.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
+# HD Wallets
```
Comment (fedekunze) [inline docs/cosmos-hub/hd-wallets.md]: this should live in the SDK docs

File: docs/cosmos-hub/keys.md [hunk 10]
```diff
@@ -0,0 +1,10 @@
+# Keys
+
+See the [Tendermint specification](https://github.com/tendermint/tendermint/blob/master/docs/spec/blockchain/encoding.md#public-key-cryptography) for how we work with keys.
+
+See `gaiacli keys --help`.
+
+Also see the [testnet
+tutorial](https://github.com/cosmos/cosmos-sdk/tree/develop/cmd/gaia/testnets).
```
Comment (fedekunze) [inline docs/cosmos-hub/keys.md]: this link is broken

File: docs/cosmos-hub/keys.md [hunk 11]
```diff
@@ -0,0 +1,10 @@
+# Keys
```
Comment (fedekunze) [inline docs/cosmos-hub/keys.md]: this should live in the SDK docs

File: docs/cosmos-hub/what-is-gaia.md [hunk 12]
```diff
@@ -0,0 +1,22 @@
+# What is Gaia?
```
Comment (fedekunze) [inline docs/cosmos-hub/what-is-gaia.md]: this section should also go into the repo's main `README`, people get confused between Gaia and the Cosmos Hub

**Comment by fedekunze** [comment]
@alessio can you copy the contribution guidelines and issue/PR templates from the SDK ?

File: docs/cosmos-hub/deploy-testnet.md (hunk 1 above)
Comment (alessio) [reply docs/cosmos-hub/deploy-testnet.md]: Ah! It must be :)

**Comment by gamarin2** [comment]
@alessio 

you can move everything in `/docs/cosmos-hub` to `/docs`. We don't need the `cosmos-hub` sub-folder anymore

File: docs/cosmos-hub/what-is-gaia.md (hunk 12 above)
Comment (alessio) [reply docs/cosmos-hub/what-is-gaia.md]: Absolutely, we need a clear explainer in the README

File: docs/cosmos-hub/README.md [hunk 13]
```diff
@@ -0,0 +1,24 @@
+# Cosmos Hub Documentation
+
+Welcome to the documentation of the **Cosmos Hub application: `gaia`**.
+
```
Comment (fedekunze) [inline docs/cosmos-hub/README.md]: Missing section `./what-is-gaia.md`

File: docs/cosmos-hub/deploy-testnet.md [hunk 14]
```diff
@@ -0,0 +1,261 @@
+# Deploy Your Own Gaia Testnet
+
+This document describes 3 ways to setup a network of `gaiad` nodes, each serving a different usecase:
+
+1. Single-node, local, manual testnet
+2. Multi-node, local, automated testnet
+3. Multi-node, remote, automated testnet
+
+Supporting code can be found in the [networks directory](https://github.com/cosmos/cosmos-sdk/tree/develop/networks) and additionally the `local` or `remote` sub-directories.
```
Comment (fedekunze) [inline docs/cosmos-hub/deploy-testnet.md]:
```suggestion
Supporting code can be found in the [networks directory](https://github.com/cosmos/cosmos-sdk/tree/master/networks) and additionally the `local` or `remote` sub-directories.
```

File: docs/cosmos-hub/deploy-testnet.md [hunk 15]
```diff
@@ -0,0 +1,261 @@
...
+
+# Now its safe to start `gaiad`
+gaiad start
+```
+
+This setup puts all the data for `gaiad` in `~/.gaiad`. You can examine the genesis file you created at `~/.gaiad/config/genesis.json`. With this configuration `gaiacli` is also ready to use and has an account with tokens (both staking and custom).
+
+## Multi-node, Local, Automated Testnet
+
+From the [networks/local directory](https://github.com/cosmos/cosmos-sdk/tree/develop/networks/local):
```
Comment (fedekunze) [inline docs/cosmos-hub/deploy-testnet.md]:
```suggestion
From the [networks/local directory](https://github.com/cosmos/cosmos-sdk/tree/master/networks/local):
```

File: docs/cosmos-hub/deploy-testnet.md [hunk 16]
```diff
@@ -0,0 +1,261 @@
...
+If you have multiple binaries with different names, you can specify which one to run with the BINARY environment variable. The path of the binary is relative to the attached volume. For example:
+
+```
+# Run with custom binary
+BINARY=gaiafoo make localnet-start
+```
+
+## Multi-Node, Remote, Automated Testnet
+
+The following should be run from the [networks directory](https://github.com/cosmos/cosmos-sdk/tree/develop/networks).
```
Comment (fedekunze) [inline docs/cosmos-hub/deploy-testnet.md]:
```suggestion
The following should be run from the [networks directory](https://github.com/cosmos/cosmos-sdk/tree/master/networks).
```

File: docs/cosmos-hub/genesis.md [hunk 17]
```diff
@@ -0,0 +1,332 @@
...
+The `genesis_time` is defined at the top of the genesis file. It is a `UTC` timestamps which specifies when the blockchain is due to start. At this time, genesis validators are supposed to come online and start participating in the consensus process. The blockchain starts when more than 2/3rd of the genesis validators (weighted by voting power) are online. 
+
+```json
+"genesis_time": "2019-03-13T17:00:00.000000000Z",
+```
+
+The `chain_id` is a unique identifier for your chain. It helps differentiate between different chains using the same version of the software.
+
+```json
+"chain_id": "cosmoshub-1",
```
Comment (fedekunze) [inline docs/cosmos-hub/genesis.md]:
```suggestion
"chain_id": "cosmoshub-2",
```

File: docs/cosmos-hub/genesis-state.md [hunk 18]
```diff
@@ -0,0 +1,60 @@
+# Gaia Genesis State
```
Comment (fedekunze) [inline docs/cosmos-hub/genesis-state.md]: what's the difference between this file and `docs/cosmos-hub/genesis.md` ? if this one is Gaia specific, I guess we can move the other to the SDK docs and link it from here

File: docs/cosmos-hub/installation.md [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+### Install the binaries
+
+Next, let's install the latest version of Gaia. Make sure you `git checkout` the correct [released version](https://github.com/cosmos/gaia/releases).
+
+::: warning
+For the mainnet, make sure your version is greater than `v0.33.0`
+:::
+
+::: warning
+Older releases up to 0.34.x are available from the [Cosmos SDK releases page](https://github.com/cosmos/cosmos-sdk/releases).
```
Comment (fedekunze) [inline docs/cosmos-hub/installation.md]:
```suggestion
Older releases of the Cosmos SDK up to 0.34.x are available on the [releases page](https://github.com/cosmos/cosmos-sdk/releases).
```

File: docs/cosmos-hub/installation.md [hunk 20]
```diff
@@ -0,0 +1,74 @@
...
+
+```bash
+$ gaiad version --long
+$ gaiacli version --long
+```
+
+`gaiacli` for instance should output something similar to:
+
+```
+cosmos-sdk: 0.33.0
```
Comment (fedekunze) [inline docs/cosmos-hub/installation.md]: not related, but should the version also print the gaia release version ? @alessio

File: docs/cosmos-hub/upgrade-node.md [hunk 21]
```diff
@@ -0,0 +1,93 @@
...
+cd $GOPATH/src/github.com/cosmos/cosmos-sdk
+git fetch --all && git checkout <new_version>
+make tools install
+```
+
+::: tip
+*NOTE*: If you have issues at this step, please check that you have the latest stable version of GO installed.
+:::
+
+See the [testnet repo](https://github.com/cosmos/testnets) for details on which version is needed for which public testnet, and the [SDK release page](https://github.com/cosmos/cosmos-sdk/releases) for details on each release.
```
Comment (fedekunze) [inline docs/cosmos-hub/upgrade-node.md]:
```suggestion
See the [testnet repo](https://github.com/cosmos/testnets) for details on which version is needed for which public testnet, and the [Gaia](https://github.com/cosmos/gaia/releases) and [SDK](https://github.com/cosmos/cosmos-sdk/releases) release pages for details on each release.
```

File: docs/cosmos-hub/upgrade-node.md [hunk 22]
```diff
@@ -0,0 +1,93 @@
...
+
+### Fetching from a Trusted Source
+
+If you are joining the mainnet, fetch the genesis from the [mainnet repo](https://github.com/cosmos/launc). If you are joining a public testnet, fetch the genesis from the appropriate testnet in the [testnet repo](https://github.com/cosmos/testnets). Otherwise, fetch it from your trusted source. 
+
+Save the new genesis as `new_genesis.json`. Then replace the old `genesis.json` with `new_genesis.json`
+
+```bash
+cd $HOME/.gaiad/config
+cp -f genesis.json new-_enesis.json
```
Comment (fedekunze) [inline docs/cosmos-hub/upgrade-node.md]:
```suggestion
cp -f genesis.json new_genesis.json
```

File: docs/cosmos-hub/upgrade-node.md [hunk 23]
```diff
@@ -0,0 +1,93 @@
...
+cd $HOME/.gaiad/config
+gaiad export --for-zero-height --height=<export-height> > new_genesis.json
+```
+
+The command above take a state at a certain height `<export-height>` and turns it into a new genesis file that can be used to start a new network. 
+
+Then, replace the old `genesis.json` with `new_genesis.json`.
+
+```bash
+cp -f genesis.json new-_enesis.json
```
Comment (fedekunze) [inline docs/cosmos-hub/upgrade-node.md]:
```suggestion
cp -f genesis.json new_genesis.json
```

File: docs/cosmos-hub/upgrade-node.md [hunk 24]
```diff
@@ -0,0 +1,93 @@
...
+mv new_genesis.json genesis.json
+```
+
+At this point, you might want to run a script to update the exported genesis into a genesis that is compatible with your new version. For example, the attributes of a the `Account` type changed, a script should query encoded account from the account store, unmarshall them, update their type, re-marshal and re-store them. You can find an example of such script [here](https://github.com/cosmos/cosmos-sdk/blob/develop/contrib/export/v0.33.x-to-v0.34.0.py).
```
Comment (fedekunze) [inline docs/cosmos-hub/upgrade-node.md]:
```suggestion
At this point, you might want to run a script to update the exported genesis into a genesis that is compatible with your new version. For example, the attributes of a the `Account` type changed, a script should query encoded account from the account store, unmarshall them, update their type, re-marshal and re-store them. You can find an example of such script [here](https://github.com/cosmos/cosmos-sdk/blob/master/contrib/export/v0.33.x-to-v0.34.0.py).
```

File: docs/cosmos-hub/validators/kms/kms.md [hunk 25]
```diff
@@ -0,0 +1,34 @@
+# KMS - Key Management System
```
Comment (fedekunze) [inline docs/cosmos-hub/validators/kms/kms.md]: should this also live in the SDK ? @alessio @sabau

File: docs/cosmos-hub/validators/kms/kms_ledger.md [hunk 26]
```diff
@@ -0,0 +1,113 @@
+# Setting up Tendermint KMS + Ledger
```
Comment (fedekunze) [inline docs/cosmos-hub/validators/kms/kms_ledger.md]: ditto, the kms should work for every chain

**Comment by alessio** [comment]
@gamarin2 dixit:

> you can move everything in /docs/cosmos-hub to /docs. We don't need the cosmos-hub sub-folder anymore

Done!

**Review by fedekunze — COMMENTED** [review]
second pass

**Review by gamarin2 — APPROVED** [review]
Looks good to me

**Pull request merged**